dask/task_spec.py

```diff
@@ -0,0 +1,742 @@
+            )
+            assert len(new_args_l) == len(subgraph.inkeys)
+            sub_dsk.update(dict(zip(subgraph.inkeys, new_args_l)))
+            func = sub_dsk[subgraph.outkey].inline(sub_dsk)
+            dct = DictOfTasks(None, {k: Alias(k, k) for k in func.dependencies})
+            return Task(key=k, func=func, args=[dct])
+        else:
+            new_args = tuple(
+                convert_old_style_task(None, a, all_keys, only_refs) for a in args
+            )
+            if only_refs:
+                # this can't be a literal since the literal wouldn't execute
+                # new_args in case there were runnable tasks or other literals
+                # See test_convert_task_only_ref
+                return Task(k, identity, (func, *new_args))
+            return Task(key=k, func=func, args=new_args)
+    try:
+        if arg in all_keys:
+            return Alias(arg, arg)
+    except TypeError:
+        # Unhashable
+        pass
+
+    if isinstance(arg, (list, tuple, set, frozenset)):
+        rv: BaseTask
+        if is_namedtuple_instance(arg):
+            if hasattr(arg, "__getnewargs_ex__"):
+                new_args, kwargs = arg.__getnewargs_ex__()
+                kwargs = DictOfTasks(
+                    None,
+                    {
+                        k: convert_old_style_task(None, v, all_keys, only_refs)
+                        for k, v in kwargs.items()
+                    },
+                )
+            elif hasattr(arg, "__getnewargs__"):
+                new_args = arg.__getnewargs__()
+                kwargs = {}
+
+            args_converted = SequenceOfTasks(
+                None,
+                [
+                    convert_old_style_task(None, a, all_keys, only_refs)
+                    for a in new_args
+                ],
+                typ=type(new_args),
+            )
+            from functools import partial
+
+            rv = Task(
+                k, partial(_instantiate_named_tuple, type(arg)), args_converted, kwargs
+            )
+        else:
+            rv = SequenceOfTasks(
+                k,
+                [convert_old_style_task(None, a, all_keys, only_refs) for a in arg],
+                typ=type(arg),
+            )
+        return rv.propagate_literal()
+    elif isinstance(arg, dict):
+        new_dsk = convert_old_style_dsk(arg, all_keys=all_keys, only_refs=True)
+
+        return DictOfTasks(None, new_dsk).propagate_literal()
+
+    elif isinstance(arg, WrappedKey):
+        return Alias(arg.key, arg.key)
+    else:
+        return LiteralTask(k, arg)
+
+
+def convert_old_style_dsk(dsk, all_keys=None, only_refs=False):
+    if not all_keys:
+        all_keys = set(dsk)
+    new_dsk = {}
+    for k, arg in dsk.items():
+        t = convert_old_style_task(k, arg, all_keys, only_refs)
+        if isinstance(t, Alias) and t.key == k:
+            continue
+        new_dsk[k] = t
+    return new_dsk
+
+
+class KeyRef:
```

Comment:
TODO: Maybe this should inherit from WrappedKey or be replaced by it entirely.

Reply:
For me, calling this class `TaskRef` would make more sense since it does reference a `Task` via its key.
```suggestion
class TaskRef:
```

Reply:
I'm actually leaning towards nuking this class entirely in favor of `WrappedKey` (keeping that for backwards compat) but I'm happy to have `TaskRef` as an alias